hardware/battery/battery_management_system_protocol.ksy

```diff
@@ -0,0 +1,223 @@
+meta:
+  id: battery_management_system_protocol
+  title: Communication protocol of smart battery management systems from LLT power
+  license: CC0-1.0
+  ks-version: 0.8
+  endian: be
+
+doc: |
+  Many modern general purpose BMS include a UART/Bluetooth based communication interface.
+  After sending read requests they respond with various information's about the battery state in
+  a custom binary format.
+
+doc-ref: https://www.lithiumbatterypcb.com/Protocol%20English%20Version.rar
+
+seq:
+  - id: magic_start
+    contents: [0xdd]
+  - id: cmd
+    type: u1
```

Comment:
Should it be a enum?

Reply:
Could be, but can enums deal with 'other' values? I think this would throw errors here.

Reply:
See https://github.com/kaitai-io/kaitai_struct/issues/778 for nuances about each of the supported language.

Reply:
I would not want to rely on that, If there is currently no consistent behavior across supported languages.

Reply:
The values `0xa5` and `0x5a` used in this switch

https://github.com/kaitai-io/kaitai_struct_formats/blob/d96be77a2b6b6e6aa7f8d8fdbc0188bc0abd7be3/hardware/battery/battery_management_system_protocol.ksy#L22-L28

should really be part of an enum - this is exactly what [enums](https://doc.kaitai.io/user_guide.html#enums) are intended for.

> I would not want to rely on that, if there is currently no consistent behavior across supported languages.

That's fine, but in that case please attach <code>-affected-by: <a href="https://github.com/kaitai-io/kaitai_struct/issues/778">778</a></code> on the `cmd` field:

```diff
   - id: cmd
     type: u1
+    -affected-by: 778
```

and at least resolve the integer keys from the enum:

```diff
   - id: body
     type:
       switch-on: cmd
       cases:
-        0xa5: read_req
-        0x5a: write_req
+        commands::read.to_i: read_req
+        commands::write.to_i: write_req
         _: response(cmd)
```

```diff
+enums:
+  commands:
+    0xa5: read
+    0x5a: write
```

Having the integer keys centralized in the enum is good, because it is less error-prone and transparent to reference the enum values like `commands::read` than repeating integer keys like `0xa5` or `165` at many places.

Reply:
Done: 064f90a